### ISSUE TITLE
ESS_GUI: Provide waterfall plot capability

### ISSUE THREAD

**Issue by smk78** [SasView/sasview#1549]
A couple of Users recently have been disappointed that SasView does not currently generate so-called 'waterfall plots', like this:

![waterfall](https://user-images.githubusercontent.com/10938679/82915334-7f565180-9f68-11ea-97f8-4427203bb554.png)

This is apparently possible with MatPlotLib using the PolyCollection class. See, for example: http://austringer.net/wp/index.php/2011/05/20/plotting-a-dolphin-biosonar-click-train/

Notice that implementing this would require some sort of dialog to capture the information and legend for the non-scattering axis (time in the WAXS example above, but could conceivably be anything).

Mantid has a simplistic (non-isometric view) implementation (mantidplot.waterfallPlot) whereby the displayed data from the 2nd and subsequent datasets are simply displaced in X and Y by offsets, however that means those data cannot be visually related to the X or Y axes, whereas, in the examples above, they can.

**Comment by rozyczko** [comment]
Yes, we have been considering adding more sophisticated plotting, likely using the ESRF's `silx` library. `Silx`, however, does not provide waterfall plots so we will need to look for something different, like PolyCollection.
Definitely very useful.

**Comment by butlerpd** [comment]
agreed .. a lot of plotting functionality improvements would be nice (though until we can do simple things like polydispersity and resolution right maybe not the top immediate priority ☹️  But I think this will be important .. could be an option for the batch fitting for example?!
